packages/checkout/blocks-registry/register-checkout-block.ts

```diff
@@ -2,80 +2,39 @@
  * External dependencies
  */
 import { registerBlockComponent } from '@woocommerce/blocks-registry';
-import { registerExperimentalBlockType } from '@woocommerce/block-settings';
 
 /**
  * Internal dependencies
  */
 import type { CheckoutBlockOptions } from './types';
-import {
-	assertBlockName,
-	assertOption,
-	assertBlockComponent,
-	assertValidArea,
-} from './utils';
+import { assertBlockName, assertOption, assertBlockComponent } from './utils';
 import { registeredBlocks } from './registered-blocks';
 
 /**
  * Main API for registering a new checkout block within areas.
  */
 export const registerCheckoutBlock = (
-	blockName: string,
 	options: CheckoutBlockOptions
 ): void => {
-	assertBlockName( blockName );
-	assertOption( options, 'areas', 'array' );
+	assertOption( options, 'metadata', 'object' );
```

Comment:
Should we run any assertion on `parent`?

Reply:
Added in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4684/commits/61091b8fc3779b5073c174f8b417c57ab6ce59df